src/custom_logging/filters.py

```diff
@@ -9,18 +9,19 @@
 
 
 class CustomFilter(Filter):
-    def __init__(self, capture_list=None, *args, **kwargs):
+    def __init__(self, capture_list=None, default_values=None, *args, **kwargs):
         super().__init__()
         if not isinstance(capture_list, (list, tuple)):
             raise CaptureListValidationException
 
         self.capture_list = capture_list
+        self.default_values = default_values or {}
 
     def filter(self, record: LogRecord) -> bool:
         for capture_in, capture_out in self.capture_list:
             # FIXME: default value should be differ by its type
             setattr(
-                record, capture_out, getattrd(local_thread, capture_in, None) or "-"
+                record, capture_out, getattrd(local_thread, capture_in, None) or self.default_values.get(capture_out, None)
```

Comment:
```suggestion
                record, capture_out, getattrd(local_thread, capture_in, None) or self.default_values.get(capture_out, "-")
```
I think placeholder(-) should be given, in case of `default_values` is not fully provided for every formatting field. Otherwise, it can break logging app(logstash, fluentbit, etc.)'s tokenizer(ex. dissect)

Reply:
Hmm replace None by "-", it will be workable.